controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -814,36 +813,33 @@ func (r *StorageClassClaimReconciler) reconcileCephFilesystemSubVolumeGroup() er
 		},
 	}
 	if err := r.get(&cephFilesystem); err != nil {
-		return fmt.Errorf("error fetching CephFilesystemList. %+v", err)
+		return fmt.Errorf("error fetching CephFilesystem. %+v", err)
 	}
 
 	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.cephFilesystemSubVolumeGroup, func() error {
 		if err := r.own(r.cephFilesystemSubVolumeGroup); err != nil {
 			return err
 		}
 		deviceClass := r.storageProfile.DeviceClass
-		cephFileSystemName := ""
-		dataPoolName := ""
+		var dataPool rookCephv1.NamedPoolSpec
 		for i := range cephFilesystem.Spec.DataPools {
 			if cephFilesystem.Spec.DataPools[i].DeviceClass == deviceClass {
-				cephFileSystemName = cephFilesystem.Name
-				dataPoolName = cephFilesystem.Spec.DataPools[i].Name
+				dataPool = cephFilesystem.Spec.DataPools[i]
 				break
 			}
 		}
-
-		if cephFileSystemName == "" {
-			return fmt.Errorf("no CephFileSystem found in the cluster with performance profile %s", r.storageClassClaim.Spec.StorageProfile)
+		if dataPool.Name == "" {
```

Comment:
In accordance with the last comment, lets change this to a `nil` check
```suggestion
		if dataPool == nil {
```